api-server/src/server/boot/user.js

```diff
@@ -468,6 +475,48 @@ function postResetProgress(req, res, next) {
   );
 }
 
+function createPostResetProjectProgress(app) {
+  return async function postResetProjectProgress(req, res, next) {
+    try {
+      const User = app.models.User;
+      const userId = req.user.id;
+      const { dashedBlockName } = req.params;
+
+      const toResetChallengeIds = getChallengeIdsForBlock(dashedBlockName);
+
+      const result = await User.updateAll(
+        {
+          id: userId,
+          'completedChallenges.id': { in: toResetChallengeIds }
```

Comment:
As far as I am aware, this is unnecessary, because there is only one user matching the `id: `userId`.

Reply:
Thank you for your feedback, @ShaunSHamilton. My intent with adding the `completedChallenges.id` in the query was to ensure that we only attempt to update the user document if there are actually completed challenges that match the IDs we're looking to reset. This was done to prevent unnecessary write operations, as MongoDB reports a document as "matched" and potentially "modified" even if the update operation does not result in any actual change to the document.

From my local tests, when using just the `userId` in the query, MongoDB indicated a match (result count = 1) even if there were no `completedChallenges` that matched the `toResetChallengeIds`. This implies that a write operation may still be reported even when no changes were made. By adding `"completedChallenges.id": { $in: toResetChallengeIds }`, I aimed to ensure that we only match and update user documents where relevant changes are necessary.

If you're aware of a more efficient approach or if my understanding is incorrect, I would very much appreciate your insights.

Reply:
Thank you for clarifying. That makes sense. So, provided the correct index is used with the query, this should work well.